pulsar-client-api/src/main/java/org/apache/pulsar/client/api/Reader.java

```diff
@@ -155,6 +157,33 @@
      */
     void seek(long timestamp) throws PulsarClientException;
 
+    /**
+     * Reset the subscription associated with this consumer to a specific message id or message publish time.
+     * <p>
+     * The Function input is topic+partition.
+     * <p>
+     * The return value is the seek position/timestamp of the current partition.
+     * <p>
+     * If returns null, the current partition will not do any processing.
+     * @param function
+     * @throws PulsarClientException
+     */
+    void seek(Function<String, Object> function) throws PulsarClientException;
```

Comment:
I am not sure that `Object` is a good return type.
Just by reading the docs, which data type I have to return ?

Reply:
We should also describe what happens in case of a RuntimeException thrown by the function.
1) is the exception rethrown, wrapped by some   subclass of PulsarClientException ?
2) if the exception is thrown not at the very first topic, what happens to the topics that have already been processed by the function ?

we should that this behaviour in the docs (and have test cases)

(this comment applies to seekAsync as well)

Reply:
Thanks Review.
If I divide the time and messageId into two interfaces, the flexibility is not so good. Now we can seek time or MessageId at the same time in Function.
Therefore, I think it can be explained through the documentation. Now if you return an object other than the timestamp and MessageId, an exception will be thrown, and the error will prompt that only timestamp and MessageId are supported

In addition, for throwing exceptions, I will add in the documentation and add corresponding unit tests.